CMakeLists.txt

```diff
@@ -116,11 +122,44 @@ if (RH_STANDALONE_PROJECT)
     target_sources_local(rh PUBLIC .clang-tidy)
 else()
     # creates a library robin_hood which is an interface (header files only)
-    add_library(robin_hood INTERFACE)
-    add_library(robin_hood::robin_hood ALIAS robin_hood)
+
+    set(LIB_NAME "robin_hood")
+    set(LIB_NAMESPACE "robin_hood")
+
+    add_library(${LIB_NAME} INTERFACE)
+    add_library(${LIB_NAMESPACE}::${LIB_NAME} ALIAS ${LIB_NAME})
 
     target_include_directories(robin_hood INTERFACE
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/include>
-        $<INSTALL_INTERFACE:src/include>
+        $<INSTALL_INTERFACE:include>
+    )
+
+    include(GNUInstallDirs)
+
+    install(
+        TARGETS ${LIB_NAME}
+        EXPORT ${LIB_NAME}-targets
+    )
+
+    install(
+        FILES src/include/robin_hood.h
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+    )
+
+    configure_file(
+        cmake/config.cmake.in
+        ${LIB_NAME}Config.cmake
+        @ONLY
+    )
+
+    install(
+        FILES ${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME}Config.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_NAME}
+    )
+
+    install(
+        EXPORT ${LIB_NAME}-targets
+        NAMESPACE ${LIB_NAMESPACE}::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_NAME}
```

Comment:
Install header-only library in non standalone mode by default is are bad idea in my opinion.
Please add an option to disable.

Reply:
What do these lines have to do with standalone mode? Unless you're talking about `RH_STANDALONE_PROJECT`, whose default value is exactly what it was before, except that now it can be overridden as part of the invocation to cmake.

Reply:
You added "install" if RH_STANDALONE_PROJECT == OFF, this changes the behavior.

Reply:
You have to install the library or else it can't be consumed by other cmake projects

Reply:
If RH_STANDALONE_PROJECT == ON - yes.
If RH_STANDALONE_PROJECT == OFF I use add_subdirectory(path/to/robin-hood-hashing) to add library in my cmake project. In this case, installation of headers is not required to use the library. robin-hood-hashing - is header-only library.

Reply:
An `install` command in CMake does not immediately install the files; it adds instructions on how to install the files. If you consume the target via `add_subdirectory`, then the `install` commands are meaningless for you because you'll never install the target and thus the commands will never actually be run. However, if you do install the library via `cmake --build build --target install`, then those `install` commands will be used to determine how to install the target to some external install directory on disk (this is very common on Linux). The library can then be consumed via `find_library` which requires a config file.

Currently, the library is unusable via `find_library` because it's missing install commands and a config file. This patch adds those. This patch has no impact on people who consume this library via `add_subdirectory`.

Reply:
CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.5)
project(test LANGUAGES CXX)
add_subdirectory(3rdparty/robin-hood-hashing)
```

Cli:
```
$ mkdir 3rdparty && cd 3rdparty
$ git clone https://github.com/Ryan-rsm-McKenzie/robin-hood-hashing.git
$ cd ../..
$ mkdir build && cd build
$ cmake -DCMAKE_INSTALL_PREFIX=/Users/phprus/Devel/install ..
$ make install
Install the project...
-- Install configuration: ""
-- Installing: /Users/phprus/Devel/install/include/robin_hood.h
-- Installing: /Users/phprus/Devel/install/lib/cmake/robin_hood/robin_hoodConfig.cmake
-- Installing: /Users/phprus/Devel/install/lib/cmake/robin_hood/robin_hood-targets.cmake
```

What is it files on add_subdirectory case???

For the find_library case, the install commands should be in the RH_STANDALONE_PROJECT == ON branch.

Reply:
These files describe your transitive dependencies. If someone uses your library, they will also need your dependencies. These files describe those dependencies. This is working exactly as intended.

Reply:
Furthermore, the `RH_STANDALONE_PROJECT=ON` is the development branch and is clearly not intended for consumption by third parties.

Reply:
In my example, NOBODY is using the library, but the unneeded files in this scenario were created by the install commands.

Please, see https://github.com/Cyan4973/xxHash/blob/dev/cmake_unofficial/CMakeLists.txt#L121 for example.
install commands used only if "NOT XXHASH_BUNDLED_MODE"
XXHASH_BUNDLED_MODE option is equivalent (NOT RH_STANDALONE_PROJECT)